src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -526,7 +544,7 @@ void OnOffServer::updateOnOffTimeCommand(chip::EndpointId endpoint)
     if (isOn) // OnOff On case
     {
         // Restart Timer
-        emberEventControlSetDelayMS(configureEventControl(endpoint), UPDATE_TIME_MS);
+        emberEventControlSetDelayMS(configureEventControl(endpoint), (uint32_t)(calculateWaitTimeMS(enterTime, chip::System::Clock::Milliseconds32(UPDATE_TIME_MS)).count()));
```

Comment:
This still has the same issue: we are compensating for the time spent in `updateOnOffTimeCommand` but not for how late `updateOnOffTimeCommand` ran compared to when it was scheduled for.  Is that purposeful (e.g. due to not wanting to store the "when we scheduled for" value)?  Might be worth documenting....